File: examples1/tree.go
package main

func main() {
	t := NewLinkedList(12)
	println(t.value)
}

type LinkedList[T any] struct {
	value T
	next  *LinkedList[T]
}

func NewLinkedList[T any](value T) LinkedList[T] {
	return LinkedList[T]{value: value, next: nil}
}

File: examples1/tree.rs
fn main() {
    let t = LinkedList::new(12);
    println!("{}", t.value);
}

struct LinkedList<T> {
    value: T,
    next: Option<Box<Self>>,
}

impl<T> LinkedList<T> {
    fn new(val: T) -> Self {
        Self { value, next: None }
    }
}
